app.rb

```diff
@@ -0,0 +1,164 @@
+      print "\n________________________________________\n"
+      puts "\n[ Thanks for using SchoolApp, Goodbye! ]"
+      print "\________________________________________\n\n"
+      exit
+    else
+      puts "\nWrong choice! plese try again\n\n"
+      menu_choice
+    end
+  end
+
+  # Create a method that shows the list of books
+  def show_books
+    @books.each do |book, i|
+      puts "#{i}Id: #{book.id}\nTitle: #{book.title}\nAuthor: #{book.author}"
+    end
+    puts "\n\n"
+  end
+
+  # Create a method that shows the list of people
+  def show_people
+    @people.each do |person, i|
+      puts "\n#{i}_____[#{person.class.name}]_____\nId: #{person.id}\nName: #{person.name} \nAge: #{person.age}\n\n"
+    end
+    "\n\n"
+  end
+
+  # Create a method that adds a student
+  def add_student
+    print 'Enter the age of the student: '
+    age = gets.chomp.to_i
+    print 'Enter the name of the student: '
+    name = gets.chomp
+    print 'The student has a parent permission? (y/n): '
+    parent_permission = gets.chomp.to_s
+    case parent_permission
+    when 'y'
+      true
+    when 'n'
+      false
+    else
+      puts "\nWrong choice! Please try again.\n\n"
+      menu_choice
+      "\n\n"
+    end
+    @people << Student.new(age, name, parent_permission)
+    puts "\nStudent was successfully added!\n\n"
+  end
+
+  # Create a method that adds a teacher
+  def add_teacher
+    print 'Enter the age of the teacher: '
+    age = gets.chomp.to_i
+    print 'Enter the name of the teacher: '
+    name = gets.chomp
+    print 'Enter the specialization of the teacher: '
+    specialization = gets.chomp
+    @people << Teacher.new(age, name, specialization)
+    puts "\nThe teacher was added successfully!\n\n"
+  end
+
+  # Create a method that select the type of person to add.
+  def add_person
+    print "\nIf you want to add a teacher, press 1. If you want to add a student, press 2: "
+    choice = gets.chomp.to_i
+    case choice
+    when 1
+      add_student
+    when 2
+      add_teacher
+    else
+      puts "\nWrong choice!\n\n"
+    end
+    system('clear')
+    menu_choice
+  end
+
+  # Create a method that adds a book
+  def add_book
+    print 'Enter the title of the book: '
+    title = gets.chomp
+    print 'Enter the author of the book: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    print "\nThe book was added successfully!\n"
+    menu_choice
+  end
+
+  # Create a method that adds a rental
+  def add_rental
+    print "\nSelect a book from the following list by number: \n"
+    show_books
+    print 'Book Id: '
+    book_id = gets.chomp
+    print "\nSelect a person from the following list by number: \n"
+    show_people
+    print 'Person Id: '
+    person_id = gets.chomp
+    print "\nEnter the rental date format (yyyy/mm/dd): "
+    rental_date = gets.chomp
+    @rentals << Rental.new(@books[book_id.to_i], @people[person_id.to_i], rental_date)
+    puts 'Rental added successfully'
+  end
```

Comment:
Creating a rental throws an error:

![image](https://user-images.githubusercontent.com/38088818/181796554-50854cea-59ca-4608-b929-e769c1670188.png)

- Kindly look into this and make sure that I can successfully create a rental for a specific person.
- Please make sure that after creating a rental, I can successfully view the rental I created using the person id. The application should not quit after listing rentals as described in the comment above.